homeassistant/components/growatt_server/sensor.py

```diff
@@ -257,6 +258,45 @@ def update(self):
         except json.decoder.JSONDecodeError:
             _LOGGER.error("Unable to fetch data from Growatt server")
 
-    def get_data(self, variable):
+    def get_currency(self):
+        """Get the currency."""
+        return self.data.get("currency")
+
+    def get_data(self, entity_description):
         """Get the data."""
-        return self.data.get(variable)
+        variable = entity_description.api_key
+        api_value = self.data.get(variable)
+        previous_value = self.previous_values.get(variable, None)
+        return_value = api_value
+
+        # If we have a 'drop threshold' specified, then check it and correct if needed
+        if (
+            entity_description.previous_value_drop_threshold is not None
+            and previous_value is not None
+        ):
+            _LOGGER.debug(
+                "%s - Drop threshold specified (%s), checking for drop... API Value: %s, Previous Value: %s",
+                entity_description.name,
+                entity_description.previous_value_drop_threshold,
+                api_value,
+                previous_value,
+            )
+            diff = float(api_value) - float(previous_value)
```

Comment:
`api_value` can be `None`. That would raise here.

Reply:
Added a check in the if-statement above to ensure this value is not `None`